examples/binding-patterns-in-assignment/binding_patterns_in_assignment.bal

```diff
@@ -0,0 +1,30 @@
+import ballerina/io;
+
+int x = 0;
+int y = 1;
+
+type IntPair [int, int];
+
+function assign(IntPair ip) {
+    // The variable `ip` of type `IntPair` is destructured and
+    // assigned to the two module-level variables `x` and `y`.
+    [x, y] = ip;
+
+    io:println(x);
+    io:println(y);
+
+    swap();
+
+    io:println(x);
+    io:println(y);
+}
+
+function swap() {
+    // The swapping is done between `x` and `y` just by doing
+    // tuple destructuring without using a temporary variable
+    [x, y] = [y, x];
+}
+
+public function main() {
+    assign([x, y]);
+}
```

Comment:
This feature is coming from modern Java Script, I think. 

Some more cool ideas
```Ballerina
    int a;
    int b;
    int c;
    [a, b, _, c] = [...[1, 2], ...[3, 4]];
```
`1
2
4`

Reply:
```Ballerina
    int a;
    int b;
    int d;
    int e;
    int[] rest;
    [a, b, _, d, e, ...rest] = [...[1, 2], ...[3, 4], ...[5, 45, 345]];
    io:println(a);
    io:println(b);
    io:println(d);
    io:println(e);
    io:println(rest);
```

`1
2
4
5
[45,345]`

Reply:
```Ballerina
    int[] a1 = [1, 2];
    int[] b1 = [3, 4];
    [a1, b1] = [b1, a1];

    io:println(a1, b1);
```

Reply:
Added few more cases